feature/interface/holdtime/otg_tests/holdtime_test/holddown_timers_test.go

```diff
@@ -0,0 +1,670 @@
+}
+
+func OTGInterfaceDOWN(t *testing.T,
+	ate *ondatra.ATEDevice) time.Time {
+
+	p1 := ondatra.ATE(t, "ate").Port(t, "port1")
+	portStateAction := gosnappi.NewControlState()
+	timestamp := time.Now()
+
+	// make sure interface is not down
+	portStateAction.Port().Link().SetPortNames([]string{p1.ID()}).SetState(gosnappi.StatePortLinkState.DOWN)
+	ate.OTG().SetControlState(t, portStateAction)
+
+	return timestamp
+}
+
+func configureOTGBundle(t *testing.T,
+	ate *ondatra.ATEDevice,
+	top gosnappi.Config,
+	aggPorts []*ondatra.Port,
+	aggID string) {
+	t.Helper()
+	agg := top.Lags().Add().SetName(lagName)
+	lagID, _ := strconv.Atoi(aggID)
+	agg.Protocol().Static().SetLagId(uint32(lagID))
+
+	for i, p := range aggPorts {
+		port := top.Ports().Add().SetName(p.ID())
+		agg.Ports().Add().SetPortName(port.Name()).Ethernet().SetMac(ateSrc.MAC).SetName("LAGRx-" + strconv.Itoa(i))
+	}
+
+	dstDev := top.Devices().Add().SetName(agg.Name() + ".dev")
+	dstEth := dstDev.Ethernets().Add().SetName(lagName + ".Eth").SetMac(ateDst.MAC)
+	dstEth.Connection().SetLagName(agg.Name())
+	dstEth.Ipv4Addresses().Add().SetName(lagName + ".IPv4").SetAddress(ateDst.IPv4).SetGateway(dutDst.IPv4).SetPrefix(uint32(ateDst.IPv4Len))
+}
+
+func displaySummaryTable(t *testing.T,
+	preActionTS,
+	postActionTS string,
+	actualDurationInMS,
+	minToleranceInMS,
+	maxToleranceInMS int64,
+	expectedOperStatus,
+	actualOperStatus string,
+	pass bool) {
+
+	result := "FAIL"
+	if pass {
+		result = "PASS"
+	}
+
+	// Prepare the strings for output.
+	expectedDurationStr := "300ms" // Assuming this is a constant value
+	minToleranceStr := fmt.Sprintf("%dms", minToleranceInMS)
+	maxToleranceStr := fmt.Sprintf("%dms", maxToleranceInMS)
+	actualDurationStr := fmt.Sprintf("%dms", actualDurationInMS)
+
+	// Create a slice of metrics and corresponding values.
+	metrics := []string{"Pre-action TS", "Post-action TS", "Expected Duration", "Actual Duration", "Min Tolerance", "Max Tolerance", "Expected Oper Status", "Actual Oper Status", "Result"}
+	values := []string{preActionTS, postActionTS, expectedDurationStr, actualDurationStr, minToleranceStr, maxToleranceStr, expectedOperStatus, actualOperStatus, result}
+
+	// Find the maximum width for the metrics to align the values.
+	maxMetricWidth := 0
+	for _, metric := range metrics {
+		if len(metric) > maxMetricWidth {
+			maxMetricWidth = len(metric)
+		}
+	}
+
+	// Create the vertical table.
+	table := ""
+	for i, metric := range metrics {
+		table += fmt.Sprintf("%-*s: %s\n", maxMetricWidth, metric, values[i])
+	}
+
+	t.Logf("\n%s", table)
+}
+
+func flapOTGInterface(t *testing.T,
+	ate *ondatra.ATEDevice,
+	dut *ondatra.DUTDevice,
+	actionState string) (time.Time, time.Time, string, string) {
+
+	// Shut down OTG Interface
+	p1 := ondatra.ATE(t, "ate").Port(t, "port1")
+	portStateAction := gosnappi.NewControlState()
+
+	var otgStateChangeTsStr string
+
+	// TC2 Step 1 Read timestamp of last oper-status change  form DUT port-1
```

Comment:
We should not need to trust the clock on the DUT (which is triggering the requirement for NTP on the DUT).  Instead maintain a set of timestamps derived from time.Now() 

- subscribe to DUT interface operational state
- set OTG interface down and get local time stamp
-  when DUT interface operational state changes, set timestamp
- compare time DUT operational state changes to OTG interface down timestamp 

This may be less precise than using the DUT timestamp because the streaming event will have some latency, but it should be sufficent precision for this test.

Reply:
If this poses a major issue (precision is very bad, off more than 5 seconds?) alternatively, we can assert that the DUT should be 'preconfigured' with NTP to a timesource which is syncronized with the test runner in the readme.

Reply:
This makes sense but on the readme it mentions as one of the checks regarding when the interface is either triggered up or down to check oper-status last change on the DUT that is why I was including the NTP as part because if it was just a matter of checking the time before and after the trigger then the time.now() makes sense however the readme wanted to include the check of the interface last change which will have the router timestamp and to make all time stamps correct with the same timezone and other things we needed NTP. Im fine to remove NTP but with check oper-status  on interface of the router there there is a high likelihood of the timestamps being on different zones or just out of sync.

Reply:
Agreed that comparing the DUT time with an OTG time requires both the OTG and DUT to have a time synced within 300msec precisionof each other for this test to perform as intended.  

I think the README should be changed to match the proposed timestamp method (using the local test runner  timestamp for response of OTG port change and the local test runner timestamp of receipt of gNMI subscribe from DUT response as the estimate of when the DUT changed it's state.  

I agree that this approach is theoretically less precise than using NTP.  But NTP could be wrong or flaky too.  FOr example, if the device clock isn't synced with the OTG before the test starts supports, adding NTP to the DUT may still take several minutes or longer for the DUT to skew it's clock to the correct time.  (The DUT may or may not support an NTP step / burst update).  There is also some change (maybe in practice a non-issue) that the OTG clock is not synced.

Using timestamps from a single source (the test runner) eliminates these variables.  Even the test runner doesn't have to be synced for precise enough timing.

If the timing using this approach isn't reliable, we could just require that the DUT and OTG are both pre-configured to use NTP and switch back to getting timestamp from the DUT.